### PR TITLE
revove v2 suport for hosts

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: Remove support for v2 hosts
Additional Description:  Another PR of type remove_v2_support. first PR is #16274
Risk Level: LOW
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by jmarantz** [comment]
Tianyu: can you take a pass?

**Review by tyxia — CHANGES_REQUESTED** [review]
Thank you for working on this!
I think this PR is not in the direction we want: We are removing the v2 support. `v2 hidden_envoy_deprecated_hosts` and related support were removed in #16931.  But this PR is adding this field back.

We should also sync on this task in the future to avoid duplicated work :)

File: source/common/upstream/static_cluster.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ class StaticClusterImpl : public ClusterImplBase {
 private:
   // ClusterImplBase
   void startPreInit() override;
-
+  const Protobuf::RepeatedPtrField<envoy::config::core::v3::Address> hosts;
```
[the remark this answers is not in the document]
Comment (ankatare) [reply source/common/upstream/static_cluster.h:27]: @tyxia as discussed in other PR, it was confusion due to stale upstream code.

**Comment by ankatare** [comment]
@tyxia hope you are in slack for in sync for this activitiy . i will coordinate there

**Comment by tyxia** [comment]
> @tyxia hope you are in slack for in sync for this activitiy . i will coordinate there

Sure. My slack user name is tyxia.  Feel free to ping me. 

I think you can drop this PR then.

**Comment by ankatare** [comment]
@tyxia Yeah...

**Comment by ankatare** [comment]
Hi @tyxia I could see 2 username with same name ... pls suggest which one to use

image

**Pull request closed without merging**